src/createDataDictionary.m

```diff
@@ -7,29 +7,53 @@ function createDataDictionary(cfg, logFile)
     % will create empty field that you can then fill in manually in the JSON
     % file
 
+    fileName = strrep(logFile(1).filename, '.tsv', '.json');
+    fullFilename = getFullFilename(fileName, cfg);
+
+    jsonContent = setJsonContent(fullFilename, logFile);
+
     opts.Indent = '    ';
+    bids.util.jsonencode(fullFilename, jsonContent, opts);
 
-    fileName = strrep(logFile(1).filename, '.tsv', '.json');
+end
 
-    fileName = fullfile( ...
-                        cfg.dir.outputSubject, ...
-                        cfg.fileName.modality, ...
-                        fileName);
+function jsonContent = setJsonContent(fullFilename, logFile)
 
+    % transfer content of extra fields to json content
+    namesExtraColumns = returnNamesExtraColumns(logFile);
+
+    % default content for events file that will be overriddent if we are dealing
+    % with a stim file
     jsonContent = struct( ...
                          'onset', struct( ...
                                          'Description', 'time elapsed since experiment start', ...
-                                         'Unit', 's'), ...
+                                         'Units', 's'), ...
                          'trial_type', struct( ...
                                               'Description', 'types of trial', ...
                                               'Levels', ''), ...
                          'duration', struct( ...
-                                            'Description', 'duration of the event or the block', ...
-                                            'Unit', 's') ...
+                                            'Description', 'duration of the event', ...
+                                            'Units', 's') ...
                         );
 
-    % transfer content of extra fields to json content
-    namesExtraColumns = returnNamesExtraColumns(logFile);
+    if ismember('_stim', fullFilename)
+
+        samplingFrequency = nan;
```

Comment:
If we are initializing a stim file we need to specify the sampling frequency in the data dictionnary.